src/main/zapHomeFiles/hud/display.js

```diff
@@ -386,16 +386,18 @@ Vue.component('history-message-modal', {
 		replayInBrowser: function() {
 			let self = this;
 			let message = this.request;
-			utils.zapApiCall("/hud/action/recordRequest/?header=" + encodeURIComponent(message.header) + "&body=" + encodeURIComponent(message.body))
-			.then(response => response.json())
-			.then(json => {
-				if (json.requestUrl) {
-					window.top.location.href = json.requestUrl;
+			let channel = new MessageChannel();
```

Comment:
do you think we should have a function to wrap up this channel logic? idk may be too much in other frames.

Reply:
Agreed.